.github/workflows/gradle-smoke-test.yml

```diff
@@ -40,19 +31,34 @@ jobs:
               with:
                   java-version: '11'
                   distribution: 'adopt'
-            - name: Prepare sample builds
-              id: prepare
-              # Convert the sample script into a build that can be invoked by Gradle
-              # - Rename the script to build.gradle
-              # - Add a settings.gradle
+            - name: Publish current snapshot of CCUD Gradle plugin to Maven local repository
+              uses: gradle/gradle-build-action@v2
+              with:
+                  arguments: publishToMavenLocal -x signArchives
+                  build-root-directory: common-custom-user-data-gradle-plugin
+            - name: Replace version in build configuration
+              run: |
+                  # fetching current snapshot version
+                  CURRENT_SNAPSHOT_VERSION=$(awk '/version =/ {print $3;exit;}' common-custom-user-data-gradle-plugin/build.gradle)
```

Comment:
instead of using a snapshot version published to a local maven repository, can we use the composite build command line flag `--include-build`? https://docs.gradle.org/current/userguide/composite_builds.html#command_line_composite

this could remove the need for the script below that adds the `mavenLocal` to the `settings.gradle` file.

Reply:
I thought about this approach, but our instructions in the samples are to use the `apply from` syntax.
The other argument is to have Maven and Gradle steps alike.

This being said I'm ok with the included build approach, let me know your preference.

Reply:
I didn't mean for the samples, I meant for this build specifically, instead of having a multi-part publish to maven local and then consume from maven local, we could simplify the build by having a single step included build. Can you try this out and see if it simplifies this build yaml file?

Reply:
I agree with @runningcode that an included build would be more appropriate here. You should be able to eliminate `publishToMavenLocal`, locating and replacing the plugin version number, and adding `mavenLocal` as a plugin repository by simply running the build with: `./gradlew tasks --include-build ../common-custom-user-data-gradle-plugin`.

One downside is that we can't do something similar with the CCUDME, so the GitHub Actions won't be consistent between Maven and Gradle. But I think it's simpler and more idiomatic for Gradle.

Reply:
all good then, I'm preparing this!

Reply:
@bigdaz I think @jprinet is running in to a limitation where the `--include-build` approach does not make substitutions for settings plugins. :\

Reply:
if you can't get the substitution through an included build, I'm gently asking your review again on the current version ;)